src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1027,6 +1027,68 @@ impl Cage {
         }
     }
 
+    //------------------------------------FCHDIR SYSCALL------------------------------------
+    
+    pub fn fchdir_syscall(&self, fd: i32) -> i32 {
+        let mut path_string = String::new();
+        let unlocked_fd = self.filedescriptortable[fd as usize].read();
+        let mut first_iteration = true;
+
+        if let Some(filedesc_enum) = &*unlocked_fd {
+            match filedesc_enum {
+                File(normalfile_filedesc_obj) => {
+                    let mut inodenum = normalfile_filedesc_obj.inode;
+                    loop{
+                        let mut thisinode = match FS_METADATA.inodetable.get_mut(&inodenum) {
+                            Some(inode) => inode,
+                            None => {
+                                return syscall_error(Errno::ENOTDIR, "fchdir", "valid inodenum");
+                            },
+                        };
+                        
+                        match *thisinode {
+                            Inode::Dir(ref mut dir_inode) => {
+                                if let Some(parent_dir_inode) = dir_inode.filename_to_inode_dict.get("..") {
+                                    if *parent_dir_inode == (1 as usize){
+                                        if !first_iteration {
+                                            path_string.insert(0, '/');
+                                            break;
+                                        }
+
+                                        first_iteration = false;
+                                    }
+
+                                    match filenamefrominode(*parent_dir_inode, inodenum) {
+                                        Some(name) => {
+                                            path_string = name + "/" + &path_string;
+                                            inodenum = *parent_dir_inode;
+                                        },
+                                        None => return syscall_error(Errno::ENOTDIR, "fchdir", "path not found"),
```

Comment:
Since we pass an open `fd` to `fchdir`, can the case where the path is not found ever arise, assuming a valid `fd`?